src/mqtt_client/mod.rs

```diff
@@ -428,39 +624,103 @@ where
     }
 }
 
+#[repr(u8)]
+pub enum ResponseCode {
+    Ok = 0,
+    Continue = 1,
+
+    Error = 0xFF,
+}
+
+impl From<ResponseCode> for u8 {
+    fn from(code: ResponseCode) -> u8 {
+        code as u8
+    }
+}
+
 /// The payload of the MQTT response message to a settings update request.
 #[derive(Serialize)]
-pub struct SettingsResponse {
+struct Response<const N: usize> {
     code: u8,
-    msg: String<64>,
+    msg: String<N>,
 }
 
-impl SettingsResponse {
+impl<const N: usize> Response<N> {
     pub fn ok() -> Self {
         Self {
             msg: String::from("OK"),
-            code: 0,
+            code: ResponseCode::Ok as u8,
         }
     }
 
-    pub fn error(msg: String<64>) -> Self {
-        Self { code: 255, msg }
+    /// Generate a custom response with any response code.
+    ///
+    /// # Args
+    /// * `code` - The code to provide in the response.
+    /// * `msg` - The message to provide in the response.
+    pub fn custom(code: impl Into<u8>, message: &str) -> Self {
+        // Truncate the provided message to ensure it fits within the heapless String.
+        let message = if message.len() > N {
```

Comment:
It's a possibility, but it then precludes some use cases of `Response` in `handled_update()` by user software (or atleast makes it harder)

Reply:
> I don't think you need to check for that. Just indiscriminately truncate with `String::from(&message[..N])` Same in error().

This will panic if N is larger than `message.len()`, hence the bounds check first.

See https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=aadc07367bb233832d951ef314ded976

Reply:
Also, let's hold off on using a `&str`. Using the string slice prevents us from e.g. sending in a debug-formatted `miniconf::Error`, which means users may not see why a serialization/deserialization op is failing

Reply:
> This will panic if N is larger than message.len(), hence the bounds check first.

Oh. My fault. I was thinking Python!